pkg/reconciler/ingress/resources/secret_ce.go

```diff
@@ -33,7 +34,24 @@ func CalculateCertificateHash(secret *corev1.Secret) (string, error) {
 
 // MakeMirrorSecret creates a Secret object that mirrors a TLS Secret
 func MakeMirrorSecret(originSecret *corev1.Secret, certificateHash string) *corev1.Secret {
+	removeEmptyLinesSecret(originSecret)
 	return makeSecret(originSecret, certificateHash, config.IstioNamespace, map[string]string{
 		labelKeyDomainMappingSecret: labelValueDomainMappingSecret,
 	}, map[string]string{})
 }
+
+func removeEmptyLines(data []byte) []byte {
+	var cleanedLines []string
+	lines := strings.Split(string(data), "\n")
+	for _, line := range lines {
+		if strings.TrimSpace(line) != "" {
+			cleanedLines = append(cleanedLines, line)
+		}
+	}
+	return []byte(strings.Join(cleanedLines, "\n"))
+}
+
+func removeEmptyLinesSecret(secret *corev1.Secret) {
```

Comment:
Instead of modifying the provided input secret by reference, and therefore changing the original secret content, I would suggest to change the function signature to return a `*corev1.Secret`. Inside the function, you would then create new one for the mirror, something like ...
```go
return &corev1.Secret{
		ObjectMeta: secret.ObjectMeta.DeepCopy(),
		Type: secret.Type,
		Data: map[string][]byte{
			corev1.TLSCertKey:       < your modification code here >,
			corev1.TLSPrivateKeyKey: < your modification code here >,
		},
	}
```

Reply:
Thanks, makes sense, used deep copy